### PR TITLE
Step-by-step tutorial on multihost networking added

### PR DESCRIPTION
**Pull request by auzias**
Added the tutorial tweetted by [@docker](https://twitter.com/docker/status/670319436369428482) in November.

Feel to request modification.

Signed-off-by: auzias github@auzias.net

### CODE REVIEW

File: docs/multihost-networking-tutorial.md [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+    --cluster-store consul://127.0.0.1:8500
+
+Defines the URI of the distributed kv storage backend. As consul is distributed the node can
+
+As [consul](http://consul.io/) is distributed a local endpoint is used instead of the consul master. Hence the master can be changed without having to restart Docker daemon on all Swarm agents and the daemons communicate locally with the cluster-storage while the cluster-storages communicate together through the network. It also allows to select the Swarm manager after the daemon has been started.
+
+### The sh-aliases used
+In the following commands these two aliases are used:
+
+    alias ldocker='docker -H tcp://0.0.0.0:2375'
```
Comment (allencloud) [inline docs/multihost-networking-tutorial.md]: I do not think adding a command alias is a good way.
We can just achieve this by `export DOCKER_HOST=tcp://0.0.0.0:2375`.

File: docs/multihost-networking-tutorial.md [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+    user@master:~$ ldocker ps
+    CONTAINER ID        IMAGE               COMMAND                  CREATED             STATUS              PORTS                    NAMES
+    d1de6e4ee3fc        swarm               "/swarm join --addr=1"   5 seconds ago       Up 4 seconds        2375/tcp                 fervent_lichterman
+    338572b87ce9        swarm               "/swarm join --addr=1"   6 seconds ago       Up 4 seconds        2375/tcp                 mad_ramanujan
+    7083e4d6c7ea        swarm               "/swarm join --addr=1"   7 seconds ago       Up 5 seconds        2375/tcp                 naughty_sammet
+    0c5abc6075da        swarm               "/swarm join --addr=1"   8 seconds ago       Up 6 seconds        2375/tcp                 gloomy_cray
+    ab746399f106        swarm               "/swarm manage token:"   25 seconds ago      Up 23 seconds       0.0.0.0:5732->2375/tcp   swarm-master
+
+### After the discovery of the swarm members
+To check if the Docker cluster is healty and running, run the command:
```
Comment (allencloud) [inline docs/multihost-networking-tutorial.md]: Typo, healty -> healthy.

File: docs/multihost-networking-tutorial.md [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+      └ Containers: 4
+      └ Reserved CPUs: 0 / 12
+      └ Reserved Memory: 0 B / 49.62 GiB
+      └ Labels: executiondriver=native-0.2, kernelversion=3.16.0-4-amd64, operatingsystem=Debian GNU/Linux 8 (jessie), storagedriver=aufs
+    CPUs: 48
+    Total Memory: 198.5 GiB
+    Name: ab746399f106
+
+
+At this point swarm is deployed and all containers run will be run over different nodes. Sebian containers can be deploy with the command:
```
Comment (allencloud) [inline docs/multihost-networking-tutorial.md]: Typo. At this point swarm is deployed and all containers will be running over different nodes. Sebian containers can be deployed with the command:

File: docs/multihost-networking-tutorial.md [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+
+---
+## Multi-hosts network
+A network overlay is needed so all the containers can be "plugged in" into the same virtual LAN. To create this network overlay, execute:
+
+    user@master:~$ swarm-docker network create -d overlay net
+    user@master:~$ swarm-docker network ls|grep "net"
+    c96760503d06        net                 overlay
+
+**And voilà !**
```
Comment (allencloud) [inline docs/multihost-networking-tutorial.md]: voilà? 
Sorry, I can not read this.

File: docs/multihost-networking-tutorial.md [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+
+Once this overlay is created, add `--net=net` to the command `swarm-docker run --rm -it debian bash` and all your containers will be able to communicate natively as if they were on the same LAN. The default network is 10.0.0.0/24. The option `--subnet` allow to specify a subnetwork, i.e., `swarm-docker network create -d overlay --subnet 192.168.0.0/16 custom-subnet`. It then is possible to specify a static IP address to a container with the option `--ip`, i.e., `swarm-docker run --rm -it --net=custom-subnet --ip=192.168.0.13 debian bash`.
```
Comment (allencloud) [inline docs/multihost-networking-tutorial.md]: Typo, allow -> allows.

**Comment by allencloud** [comment]
Great.

File: docs/multihost-networking-tutorial.md (hunk 4 above)
Comment (auzias) [inline docs/multihost-networking-tutorial.md]: What do you mean? [_voilà_ expression](https://www.wordnik.com/words/voil%C3%A0).

File: docs/multihost-networking-tutorial.md (hunk 1 above)
Comment (auzias) [inline docs/multihost-networking-tutorial.md]: What about swarm then?

File: docs/multihost-networking-tutorial.md (hunk 4 above)
Comment (allencloud) [inline docs/multihost-networking-tutorial.md]: I mean that is this an English expression?

Comment (auzias) [inline docs/multihost-networking-tutorial.md]: Yes it is

**Comment by moxiegirl** [comment]
@auzias Thanks for the submission, glad it made it in.  I have to change some of the content to match our documentation style and the terminology for features that we use.  I'll do that today and get it back to you.  From your tweet, I assume I can congratulate you on completing your thesis? :-D

**Comment by auzias** [comment]
@moxiegirl feel free to change anything that need to be change! I'm glad it helps.
Hum... As for the thesis, not yet. Last year though! Thanks anyway ;)

**Comment by auzias** [comment]
@moxiegirl any update?

**Comment by moxiegirl** [comment]
@auzias Apologies for the pause...we transitioned in a new writer, Charles.  He'll be owning the Swarm documentation moving forward and can help you with this. @sfsmithcha

**Comment by auzias** [comment]
If I can help and ease this transition, feel free to ask and direct me :smiley:  :+1:

**Comment by sfsmithcha** [comment]
Thanks for the offer of help @auzias and thanks for your patience while I'm coming up to speed on the Swam docs & the Docker docs in general. I noticed that your tutorial covers some similar territory to this networking tutorial: https://docs.docker.com/engine/userguide/networking/get-started-overlay/ .

**Comment by auzias** [comment]
Hi @sfsmithcha ! Indeed, this is the doc I used to deploy the described Swarm environment within this tutorial. I'll check it and try to match with the documentation style and terminology if I have time within the week.

**Comment by nishanttotla** [comment]
Closing PR since docs have moved.

**Pull request closed without merging**